extensions/modules/file/src/main/java/org/exist/xquery/modules/file/Sync.java

```diff
@@ -55,183 +62,356 @@
+        final String target = args[1].getStringValue();
+        final HashMap<String, Sequence> options = getOptions(args[2]);
+
+        return startSync(target, collectionPath, options);
+    }
+
+    private HashMap<String, Sequence> getOptions(final Sequence parameter) throws XPathException {
+        HashMap<String, Sequence> options = new HashMap<>();
+        options.put(AFTER_OPT, Sequence.EMPTY_SEQUENCE);
+        options.put(PRUNE_OPT, new BooleanValue(false));
+        options.put(EXCLUDES_OPT, Sequence.EMPTY_SEQUENCE);
+
+        if (parameter.isEmpty()) {
+            return options;
+        }
+
+        final Item item = parameter.itemAt(0);
+
+        if (item.getType() == Type.MAP) {
+            final AbstractMapType optionsMap = (AbstractMapType) item;
+
+            final Sequence seq = optionsMap.get(new StringValue(EXCLUDES_OPT));
+            if (!seq.isEmpty() && seq.getItemType() != Type.STRING) {
+                throw new XPathException(this, ErrorCodes.XPTY0004,
+                        "Invalid value for option \"excludes\", expected xs:string* got " +
+                                Type.getTypeName(seq.getItemType()));
+            }
+            options.put(EXCLUDES_OPT, seq);
+
+            checkOption(optionsMap, PRUNE_OPT, Type.BOOLEAN, options);
+            checkOption(optionsMap, AFTER_OPT, Type.DATE_TIME, options);
+        } else if (parameter.itemAt(0).getType() == Type.DATE_TIME) {
+            // QUESTION(JL) LOG deprecation warning?
+            options.put(AFTER_OPT, parameter);
+        } else {
+            throw new XPathException(this, ErrorCodes.XPTY0004,
+                    "Invalid 3rd parameter, allowed are xs:dateTime or map(*) got " + Type.getTypeName(item.getType()));
+        }
+        return options;
+    }
+
+    private void checkOption(
+            final AbstractMapType optionsMap,
+            final String name,
+            final int expectedType,
+            final HashMap<String, Sequence> options
+    ) throws XPathException {
+        final Sequence p = optionsMap.get(new StringValue(name));
+
+        if (p.isEmpty()) {
+            return; // nothing to do, continue
+        }
+
+        if (p.hasMany() || p.getItemType() != expectedType) {
+            throw new XPathException(this, ErrorCodes.XPTY0004,
+                    "Invalid value for option \"" + name + "\", expected " +
+                            Type.getTypeName(expectedType) + " got " +
+                            Type.getTypeName(p.itemAt(0).getType()));
+        }
+
+        options.put(name, p);
+    }
+
+    private Sequence startSync(
+            final String target,
+            final String collectionPath,
+            final HashMap<String, Sequence> options
+    ) throws XPathException {
+        final Date startDate = options.get(AFTER_OPT).hasOne() ? ((DateTimeValue) options.get(AFTER_OPT)).getDate() : null;
+
+        boolean prune = ((BooleanValue) options.get(PRUNE_OPT)).getValue();
```

Comment:
`prune` should be final.
This will throw an NPE if the user doesn't specify `prune` in the map - this needs to be fixed. Should default to `false` presumably?